TROUBLESHOOTING.md

```diff
@@ -115,6 +136,178 @@ This error indicates that the access token used for authentication does not have
+
+1. **Request Service Principal Creation:**
+   Ask your Azure administrator to create a service principal with the following information:
+   ```
+   Application Name: Azure MCP Server - [Your Name/Team]
+   Required Permissions: 
+   - Reader role at subscription/resource group level
+   - Data plane roles for specific resources (e.g., Storage Blob Data Reader)
+   Justification: Development/testing with Azure MCP Server
+   ```
+
+2. **Configuration:**
+   Once the service principal is created, configure these environment variables:
+   ```bash
+   export AZURE_CLIENT_ID="service-principal-client-id"
+   export AZURE_CLIENT_SECRET="service-principal-secret"
+   export AZURE_TENANT_ID="your-tenant-id"
+   ```
+
+3. **Certificate-Based Authentication (Preferred):**
+   For enhanced security, request certificate-based authentication:
+   ```bash
+   export AZURE_CLIENT_ID="service-principal-client-id"
+   export AZURE_CLIENT_CERTIFICATE_PATH="/path/to/certificate.pem"
+   export AZURE_TENANT_ID="your-tenant-id"
+   ```
+
+#### Conditional Access Policy Compliance
+
+Organizations may enforce Conditional Access policies that affect authentication:
+
+**Common Policy Requirements:**
+- Device compliance (Azure AD joined devices)
+- Multi-factor authentication (MFA)
+- Location-based restrictions
+- Application-specific controls
+
+**Working with Identity Administrators:**
+
+1. **Check Policy Impact:**
+   ```
+   Questions to ask:
+   - Are there Conditional Access policies affecting my authentication?
+   - Is my device compliant with organizational policies?
+   - Do I need to use a specific authentication method?
+   - Can I get an exception for development scenarios?
+   ```
+
+2. **Policy Compliance Steps:**
+   - Ensure your device is Azure AD joined or hybrid joined
+   - Complete device compliance enrollment if required
+   - Use compliant authentication methods (avoid saved credentials in non-compliant browsers)
+   - Connect from approved network locations if location policies exist
+
+#### Resource Access in Locked-Down Environments
+
+When resources are heavily restricted:
+
+1. **Minimum Required Information to Gather:**
+   ```
+   Resource Details:
+   - Resource names and types
+   - Resource group and subscription
+   - Whether private endpoints are used
+   - Network restrictions (IP allowlists, VNet integration)
+   
+   Access Requirements:
+   - Required RBAC roles
+   - Network access requirements
+   - Authentication method preferences
+   ```
+
+2. **Escalation Path:**
+   ```
+   Level 1: Resource Administrator
+   - Resource-specific permissions
+   - RBAC role assignments
+   
+   Level 2: Network Administrator  
+   - Firewall rules and network access
+   - Private endpoint connectivity
+   
+   Level 3: Identity Administrator
+   - Conditional Access policies
+   - Service principal creation
+   ```
+
 ### AADSTS500200 error: User account is a personal Microsoft account
```

Comment:
Could you also help improve this specification section on `User account is a personal Microsoft account`?